File: test/fixtures/kotlin/post_with_data-ascii.kt
import java.io.IOException
import okhttp3.FormBody
import okhttp3.OkHttpClient
import okhttp3.Request

val client = OkHttpClient()

val formBody = FormBody.Builder()
  .add("msg1", "wow")
  .add("msg2", "such")
  .build()

val request = Request.Builder()
  .url("http://localhost:28139/echo/html/")
  .post(formBody)
  .header("Origin", "http://fiddle.jshell.net")
  .header("Accept-Encoding", "gzip, deflate")
  .header("Accept-Language", "en-US,en;q=0.8")
  .header("User-Agent", "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36")
  .header("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8")
  .header("Accept", "*/*")
  .header("Referer", "http://fiddle.jshell.net/_display/")
  .header("X-Requested-With", "XMLHttpRequest")
  .header("Connection", "keep-alive")
  .build()

client.newCall(request).execute().use { response ->
  if (!response.isSuccessful) throw IOException("Unexpected code $response")
  response.body!!.string()
}

File: test/fixtures/r/post_with_data-ascii.r
library(httr)

headers = c(
  Origin = "http://fiddle.jshell.net",
  `Accept-Encoding` = "gzip, deflate",
  `Accept-Language` = "en-US,en;q=0.8",
  `User-Agent` = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36",
  `Content-Type` = "application/x-www-form-urlencoded; charset=UTF-8",
  Accept = "*/*",
  Referer = "http://fiddle.jshell.net/_display/",
  `X-Requested-With` = "XMLHttpRequest",
  Connection = "keep-alive"
)

data = list(
  msg1 = "wow",
  msg2 = "such"
)

res <- httr::POST(url = "http://localhost:28139/echo/html/", httr::add_headers(.headers=headers), body = data, encode = "form")
